client/src/components/graphs/OrganizationalChart.js

```diff
@@ -69,15 +72,34 @@ const transitionDuration = 200
 
 const ranks = Settings.fields.person.ranks.map(rank => rank.value)
 
-const sortPositions = (positions, truncateLimit) => {
-  const allResults = [...positions].sort((p1, p2) =>
-    ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank) ? -1 : 1
-  )
+const leaderCountMap = {}
+
+const sortPositions = (uuid, positions, truncateLimit) => {
+  const allResults = [...positions]
+    .sort((p1, p2) => {
+      if (p1.role === p2.role) {
+        return ranks.indexOf(p1.person?.rank) > ranks.indexOf(p2.person?.rank)
+          ? -1
+          : 1
+      }
+      return Object.keys(PositionRole).indexOf(p1.role) >
+        Object.keys(PositionRole).indexOf(p2.role)
+        ? -1
+        : 1
```

Comment:
Why not also extract `const roles = Object.keys(PositionRole)` (like you do for `ranks`)?
And again, here you could simply:
```javascript
      return roles.indexOf(p2.role) - roles.indexOf(p1.role)
```

Although if you start adding additional sub-conditions, it's probably clearer to write it along the lines of:
```javascript
const roles = Object.keys(PositionRole)
const ranks = Settings.fields.person.ranks.map(rank => rank.value)

const sortPositions = (uuid, positions, truncateLimit) => {
  const allResults = positions
    ?.sort(
      (p1, p2) =>
        // first compares roles
        roles.indexOf(p2.role) - roles.indexOf(p1.role) ||
        // when these are equal, compare ranks
        ranks.indexOf(p2.person?.rank) - ranks.indexOf(p1.person?.rank) ||
        // when these are also equal, compare names
        p2.person?.name?.localeCompare(p1.person?.name) ||
        // last resort: uuid
        p2.person?.uuid.localeCompare(p1.person?.uuid)
    )
    .map(pos => ({ ...pos, orgUuid: uuid }))

  return truncateLimit !== undefined && truncateLimit < allResults.length
    ? allResults.slice(0, truncateLimit)
    : allResults
}
```